application/templates/deployment.yaml

```
@@ -141,6 +144,10 @@ spec:
         image: {{ $image }}
 
         imagePullPolicy: {{ .Values.deployment.image.pullPolicy }}
+        {{- if .Values.deployment.lifecycle }}
+        lifecycle:
+{{ toYaml .Values.deployment.lifecycle | indent 10 }}
```

Let's try to keep indentation levels consistent with where the templated values will end up.

```suggestion
        lifecycle:
          {{- toYaml .Values.deployment.lifecycle | nindent 10 }}
```